modules/dnn/test/test_layers_1d.cpp

```diff
@@ -480,6 +480,133 @@ INSTANTIATE_TEST_CASE_P(/*nothing*/, Layer_Scatter_Test, Combine(
 
 
 
+typedef testing::TestWithParam<tuple<std::vector<int>, std::string, int>> Layer_Reduce_Test;
+TEST_P(Layer_Reduce_Test, Accuracy_01D)
+{
+    auto reduceOperation = [](const cv::Mat& input, const std::string& operation, int axis) -> cv::Mat {
+        // Initialize result matrix
+        cv::Mat result;
+        if (shape(input).size() == 0 || shape(input).size() == 1){
+            result = cv::Mat(shape(input).size(), shape(input).data(), CV_32F);
+            int sz[1] = {1};
+            if (!shape(input).empty() && shape(input)[0] != 1){
+                result = cv::Mat(1, 1, CV_32F);
+                result = result.reshape(1, 1, sz);
+            }
+        } else {
+            if (axis == 0) {
+                result = cv::Mat::zeros(1, input.cols, CV_32F);
+            } else {
+                result = cv::Mat::zeros(input.rows, 1, CV_32F);
+            }
+        }
+
+        auto process_value = [&](float& res, float value, bool is_first) {
+            if (operation == "max") {
+                res = is_first ? value : std::max(res, value);
+            } else if (operation == "min") {
+                res = is_first ? value : std::min(res, value);
+            } else {
+                if (is_first) {
+                    if (operation == "sum" || operation == "l1" || operation == "l2"
+                        || operation == "sum_square" || operation == "mean" || operation == "log_sum"
+                        || operation == "log_sum_exp") res = 0;
+                    else if (operation == "prod") res = 1;
+                }
+
+                if (operation == "sum" || operation == "mean") res += value;
+                else if (operation == "sum_square") {
+                    if (shape(input).size() == 2 && shape(input)[0] == 1 && axis==0)
+                        res += value;
+                    else if (shape(input).size() == 2 && shape(input)[1] == 1 && axis==1)
+                        res += value;
```

Comment:
Should be square in both cases:
```python
@ost.script()
def layer_sum_square(x: ost.FLOAT[1, 4]) -> ost.FLOAT[1, 4]:
    return opset18.ReduceSumSquare(x, axes=[0], keepdims=1)

make_model_and_data(layer_sum_square, np.random.rand(1, 4).astype(np.float32))
```
```
input [[0.5488135  0.71518934 0.60276335 0.5448832 ]]
output [[0.30119628 0.51149577 0.36332366 0.29689768]]
```


```python
@ost.script()
def layer_sum_square(x: ost.FLOAT[1, 4]) -> ost.FLOAT[1, 4]:
    return opset18.ReduceSumSquare(x, axes=[0], keepdims=1)

make_model_and_data(layer_sum_square, np.random.rand(1, 4).astype(np.float32))
```
```
input [[0.5488135  0.71518934 0.60276335 0.5448832 ]]
output [[0.30119628 0.51149577 0.36332366 0.29689768]]
```

```python
@ost.script()
def layer_sum_square(x: ost.FLOAT[4, 1]) -> ost.FLOAT[4, 1]:
    return opset18.ReduceSumSquare(x, axes=[1], keepdims=1)

make_model_and_data(layer_sum_square, np.random.rand(4, 1).astype(np.float32))
```
```
input [[0.5488135 ]
 [0.71518934]
 [0.60276335]
 [0.5448832 ]]
output [[0.30119628]
 [0.51149577]
 [0.36332366]
 [0.29689768]]
 ```
 source: https://github.com/opencv/opencv_extra/blob/4.x/testdata/dnn/onnx/generate_onnx_models_with_onnxscript.py